08-rails/routes-controllers.md

```diff
@@ -97,20 +99,30 @@ So for example for:
 -  route:  `get "/books/:id", to: "books#show", as: "banana"`
 	-  Prefix: `banana` 
 	-  Path Helper: `banana_path`.  
+	
+Additionally any route beyond the standard set of CRUD routes should be named descriptively.  
+
+For example a path to mark a book as read could be:
+
+```ruby
+patch '/user/:user_id/books/:id', to: 'books#mark_read', as: 'mark_read'
+```
 
 Below is a table with example routes, prefixes and paths.  
 
 |   Route	|   Prefix	|   Path Helper	|	Example  |
 |---	|---	|---	|---	|
 |   `get "/books/new", to: "books#new", as: "new_book"`	|   `new_book`	|   `new_book_path`	|	`<%= link_to "New book", new_book_path %>`
 |   `get "/books/:id/edit", to: "books#edit", as: "edit_book"`	|   `edit_book`	|   `edit_book_path`	|  `<%= button_to "Edit #{book.title}", edit_book_path(book.id) %>`	|
-|   `post "/books", to: "books#create"`	|   `books`	|   `books_path`	| `<%= form_for @book, action: books_path, method: :post %>`	|
+|   `post "/books", to: "books#create"`, as: 'books'	|   `books`	|   `books_path`	| `<%= form_for @book, action: books_path, method: :post %>`	|
 |   `get "/books", to: "books#index"`	|   `books`	|   `books_path`	| `<%= link_to "All Books", books_path %>`  |
 |	`get "/books/:id", to: "books#show", as: "book"` | 	`book`  |	`book_path`  |	`<%= link_to "View #{book.title}", book_path(book.id) %>`	|
 |	`patch "/books/:id", to: "books#update"` | 	`book`  |	`book_path`  |	`<%= form_for @book, action: book_path(@book.id), method: :patch %>`  |
 |	`delete "/books/:id", to: "books#destroy"` | 	`book`  |	`book_path`  |	`<%= link_to "Delete", book_path(book.id) %>`  |
 
 
+Notice that the `delete` & `destroy` actions both use the `book_path`.  The path helpers are identical because the routes all include `/books/:id`.  
```

Comment:
I think we mean `update` & `destroy` here, although we could also add in `show`.